### PR TITLE
feat(Kafka): kafka support security protocol and enabled mechanisms

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  kafka support security protocol and enabled mechanisms
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  kafka support security protocol and enabled mechanisms
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccKafkaInstance_' TEST_PARALLELISM=8
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstance_ -timeout 360m -parallel 8
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== RUN   TestAccKafkaInstance_prePaid
=== PAUSE TestAccKafkaInstance_prePaid
=== RUN   TestAccKafkaInstance_withEpsId
=== PAUSE TestAccKafkaInstance_withEpsId
=== RUN   TestAccKafkaInstance_compatible
=== PAUSE TestAccKafkaInstance_compatible
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_compatible
=== CONT  TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_withEpsId
=== CONT  TestAccKafkaInstance_prePaid
--- PASS: TestAccKafkaInstance_withEpsId (565.12s)
--- PASS: TestAccKafkaInstance_compatible (928.15s)
--- PASS: TestAccKafkaInstance_prePaid (1094.09s)
--- PASS: TestAccKafkaInstance_basic (1257.34s)
--- PASS: TestAccKafkaInstance_newFormat (2110.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       2110.439s

make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccKafkaInstancesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstancesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccKafkaInstancesDataSource_basic
=== PAUSE TestAccKafkaInstancesDataSource_basic
=== CONT  TestAccKafkaInstancesDataSource_basic
--- PASS: TestAccKafkaInstancesDataSource_basic (560.27s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       689.246s
```

### CODE REVIEW

File: docs/resources/dms_kafka_instance.md [hunk 1]
```diff
@@ -169,6 +169,23 @@ The following arguments are supported:
   | 600MB | 4 |
   | 1,200MB | 8 |
 
+* `security_protocol` - (Optional, String, ForceNew) Specifies the protocol to use after SASL is enabled (`access_user`
+  and `password` are specified). Value options:
+  + **SASL_SSL**: Data is encrypted with SSL certificates for high-security transmission.
+  + **SASL_PLAINTEXT**: Data is transmitted in plaintext with username and password authentication. This protocol only
+    uses the SCRAM-SHA-512 mechanism and delivers high performance.
```
Comment (deer-hang) [inline docs/resources/dms_kafka_instance.md:176]:
```suggestion
  uses the SCRAM-SHA-512 mechanism and delivers high performance.
```

File: docs/resources/dms_kafka_instance.md [hunk 2]
```diff
@@ -169,6 +169,23 @@ The following arguments are supported:
   | 600MB | 4 |
   | 1,200MB | 8 |
 
+* `security_protocol` - (Optional, String, ForceNew) Specifies the protocol to use after SASL is enabled (`access_user`
+  and `password` are specified). Value options:
```
Comment (deer-hang) [inline docs/resources/dms_kafka_instance.md:173]:
```suggestion
  and `password` are specified). This parameter is valid and required when SASL is enabled. Value options:
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go [hunk 3]
```diff
@@ -636,6 +645,13 @@ func createKafkaInstanceWithProductID(ctx context.Context, d *schema.ResourceDat
 		createOpts.PublicIpID = publicIpIDs
 	}
 
+	createOpts.SslEnable = false
```
Comment (deer-hang) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: This line is useless.

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccKafkaInstance_' TEST_PARALLELISM=8
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccKafkaInstance_ -timeout 360m -parallel 8
=== RUN   TestAccKafkaInstance_basic
=== PAUSE TestAccKafkaInstance_basic
=== RUN   TestAccKafkaInstance_prePaid
=== PAUSE TestAccKafkaInstance_prePaid
=== RUN   TestAccKafkaInstance_withEpsId
=== PAUSE TestAccKafkaInstance_withEpsId
=== RUN   TestAccKafkaInstance_compatible
=== PAUSE TestAccKafkaInstance_compatible
=== RUN   TestAccKafkaInstance_newFormat
=== PAUSE TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_basic
=== CONT  TestAccKafkaInstance_compatible
=== CONT  TestAccKafkaInstance_withEpsId
=== CONT  TestAccKafkaInstance_newFormat
=== CONT  TestAccKafkaInstance_prePaid
--- PASS: TestAccKafkaInstance_withEpsId (561.56s)
--- PASS: TestAccKafkaInstance_compatible (672.31s)
--- PASS: TestAccKafkaInstance_prePaid (1072.51s)
--- PASS: TestAccKafkaInstance_basic (1405.37s)

--- PASS: TestAccKafkaInstance_newFormat (2277.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       2277.177s

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go]: yes, fixed

File: docs/resources/dms_kafka_instance.md [hunk 4]
```diff
@@ -169,6 +169,23 @@ The following arguments are supported:
...
+
+  Changing this creates a new instance resource.
+
+* `enabled_mechanisms` - (Optional, List, ForceNew) Specifies the authentication mechanisms to use after SASL is
+  enabled (`access_user` and `password` are specified). Value options:
+  + **PLAIN**: Simple username and password verification.
+  + **SCRAM-SHA-512**: User credential verification, which is more secure than **PLAIN**.
+  Defaults to **PLAIN**.
+
+  Changing this creates a new instance resource.
```
Comment (ShiChangkuo) [inline docs/resources/dms_kafka_instance.md]: we should add a new line after the list items.

```suggestion

    Defaults to [**PLAIN**].   Changing this creates a new instance resource.
```

**Pull request merged**